Examples/runtimes/java/DynamoDbEncryption/src/main/java/software/amazon/cryptography/examples/keyring/RawEcdhKeyringExample.java

```diff
@@ -0,0 +1,885 @@
+      .keyring(rawEcdhKeyring)
+      .allowedUnsignedAttributePrefix(unsignAttrPrefix)
+      .build();
+    tableConfigs.put(ddbTableName, config);
+
+    // Create the DynamoDb Encryption Interceptor
+    DynamoDbEncryptionInterceptor encryptionInterceptor =
+      DynamoDbEncryptionInterceptor
+        .builder()
+        .config(
+          DynamoDbTablesEncryptionConfig
+            .builder()
+            .tableEncryptionConfigs(tableConfigs)
+            .build()
+        )
+        .build();
+
+    // Create a new AWS SDK DynamoDb client using the DynamoDb Encryption Interceptor above
+    final DynamoDbClient ddb = DynamoDbClient
+      .builder()
+      .overrideConfiguration(
+        ClientOverrideConfiguration
+          .builder()
+          .addExecutionInterceptor(encryptionInterceptor)
+          .build()
+      )
+      .build();
+
+    // Get the item back from our table using the same client.
+    // The client will decrypt the item client-side, and return
+    // back the original item.
+    final HashMap<String, AttributeValue> keyToGet = new HashMap<>();
+    keyToGet.put(
+      "partition_key",
+      AttributeValue.builder().s("rawEcdhKeyringItem").build()
+    );
+    keyToGet.put("sort_key", AttributeValue.builder().n("0").build());
+
+    final GetItemRequest getRequest = GetItemRequest
+      .builder()
+      .key(keyToGet)
+      .tableName(ddbTableName)
+      .build();
+
+    final GetItemResponse getResponse = ddb.getItem(getRequest);
+
+    // Demonstrate that GetItem succeeded and returned the decrypted item
+    assert 200 == getResponse.sdkHttpResponse().statusCode();
+    final Map<String, AttributeValue> returnedItem = getResponse.item();
+    assert returnedItem
+      .get("sensitive_data")
+      .s()
+      .equals("encrypt and sign me!");
+  }
+
+  /*
+    To run this example standalone, you will need to supply your own ECC keys at:
+    - EXAMPLE_ECC_PRIVATE_KEY_FILENAME_SENDER
+    - EXAMPLE_ECC_PUBLIC_KEY_FILENAME_RECIPIENT
+    You will also need to provide the Dynamodb table name as part of the args.
+   */
+  public static void main(final String[] args) {
+    if (args.length <= 0) {
+      throw new IllegalArgumentException(
+        "To run this example, include the ddbTable in args"
+      );
+    }
+    final String ddbTableName = args[0];
+
+    RawEcdhKeyringGetItemPutItem(ddbTableName, ECDHCurveSpec.ECC_NIST_P256);
+  }
+
+  public static boolean shouldGenerateNewEccKeyPairs() {
+    File privateKeyFileSender = new File(
+      EXAMPLE_ECC_PRIVATE_KEY_FILENAME_SENDER
+    );
+    File privateKeyFileRecipient = new File(
+      EXAMPLE_ECC_PRIVATE_KEY_FILENAME_RECIPIENT
+    );
+    File publicKeyFileRecipient = new File(
+      EXAMPLE_ECC_PUBLIC_KEY_FILENAME_RECIPIENT
+    );
+
+    // If keys already exist: do not overwrite existing keys
+    return (
+      !privateKeyFileSender.exists() ||
+      !publicKeyFileRecipient.exists() ||
+      !privateKeyFileRecipient.exists()
```

Comment:
```suggestion
      !privateKeyFileSender.exists() &&
      !publicKeyFileRecipient.exists() &&
      !privateKeyFileRecipient.exists()
```
I think this --
Current change says "shouldGenerateNewEccKeyPairs if any of them don't exist"
Proposed change says "shouldGenerateNewEccKeyPairs if none of them exist"
Proposed change is safer